src/initialization/FGLinearization.h

```diff
@@ -19,25 +19,74 @@
 #ifndef FGLinearization_H_
 #define FGLinearization_H_
 
-#include "initialization/FGTrimmer.h"
 #include "math/FGStateSpace.h"
 #include <iomanip>
 #include <fstream>
 #include "models/FGAircraft.h"
 #include "models/propulsion/FGEngine.h"
 #include "models/propulsion/FGTurbine.h"
 #include "models/propulsion/FGTurboProp.h"
-#include "math/FGNelderMead.h"
 #include <stdexcept>
 #include <fstream>
 #include <cstdlib>
+#include <memory>
+#include <utility>
 
 namespace JSBSim {
 
+template <class T>
+using Vector2D = std::vector<std::vector<T>>;
+
+/** \brief Class used to create linear models from FGFDMExec instances.
+ */
 class FGLinearization
 {
+    Vector2D<double> A,B,C,D;
+    std::vector<double> x0, u0, y0;
+    std::vector<string> x_names, u_names, y_names, x_units, u_units, y_units;
+    std::string aircraft_name;
 public:
-    FGLinearization(FGFDMExec * fdmPtr, int mode);
+    /**
+     * @param fdmPtr Already configured FGFDMExec instance used to create the new linear model.
+     */
+    FGLinearization(FGFDMExec * fdmPtr);
+
+    /**
+     * Write Scicoslab source file with the state space model to a
+     * file in the current working directory.
+     */
+    void WriteScicoslab() const;
+
+    /**
+     * Write Scicoslab source file with the state space model to the given path.
+     *
+     * @param path
+     */
+    void WriteScicoslab(std::string& path) const;
+
+    /**
+     * Get the state space model matrices.
+     *
+     * @param A_ System matrix
+     * @param B_ Input matrix
+     * @param C_ Output matrix
+     * @param D_ Feedforward matrix
+     */
+    void GetStateSpace(Vector2D<double> & A_, Vector2D<double> & B_,
+                       Vector2D<double> & C_, Vector2D<double> & D_) const;
```

Comment:
I would split `GetStateSpace` in 4 separate methods rather than modifying the method parameters which I always find awkward.
In addition the 4 method names would document what is the returned matrix rather than using obscure names `A_`, `B_`, `C_` and `D_`.
```suggestion
    const Vector2D<double>& GetSystemMatrix(void) const { return A; }
    const Vector2D<double>& GetInputMatrix(void) const { return B; }
    const Vector2D<double>& GetOutputMatrix(void) const { return C; }
    const Vector2D<double>& GetFeedforwardMatrix(void) const { return D; }
```